tests/bytes.rs

```diff
@@ -728,11 +745,25 @@ fn create_input_coin_data_offset() {
                         .expect("Failed to serialize input");
 
                     let (offset, len) = tx
-                        .input_coin_predicate_offset(last_input)
+                        .input_predicate_offset(input_coin_idx)
+                        .expect("Failed to fetch offset");
+
+                    let (offset_p, _) = tx_p
+                        .input_predicate_offset(input_coin_idx)
+                        .expect("Failed to fetch offset from tx with precomputed metadata!");
+
+                    assert_eq!(offset, offset_p);
+                    assert_eq!(
+                        predicate.as_slice(),
+                        &buffer[offset..offset + len][..predicate.len()]
+                    );
+
+                    let (offset, len) = tx
+                        .input_predicate_offset(input_message_idx)
                         .expect("Failed to fetch offset");
 
                     let (offset_p, _) = tx_p
-                        .input_coin_predicate_offset(last_input)
+                        .input_predicate_offset(input_message_idx)
```

Comment:
I just realized this test is named `create_input_coin_data_offset`. We should probably rename it or split the message offsets into their own tests.

Reply:
Wouldn't be strictly false to have that as generic: https://github.com/FuelLabs/fuel-tx/pull/174/commits/7ea93d5c0711493c2fa459d04b2009785f785ae6